common/components/common/integrations/SocialMediaSignupSection.jsx

```diff
@@ -122,14 +128,23 @@ class SocialMediaSignupSection extends React.Component<Props, State> {
           const iconClass: string =
             "LogInController-socialIcon" +
             (config.iconClass ? " " + config.iconClass : "");
+
+	  var url = "";
+          if (this.state.prevPageArgs) {
+           url = "/api/login/" + app + "?prevPage=" +this.state.prevPage + "&prevPageArgs=" +    this.state.prevPageArgs 
+          } else {
+             url = "/api/login/" + app  + "?prevPage=" + this.state.prevPage
+          }	
```

Comment:
This logic can be compressed using helper functions.
```suggestion
	  const args: Dictionary<string> = {prevPage: this.state.prevPage};
	  if(this.state.prevPageArgs) {
	    args["prevPageArgs"] = this.state.prevPageArgs;
	  }
	  const url: string =  "/api/login/" + app + urlHelper.argsString(args);
```